networkx/algorithms/tests/test_graph_hashing.py

```diff
@@ -1,42 +1,624 @@
+import pytest
 import networkx as nx
 
 
-def test_empty_graph_hash():
-    G1 = nx.empty_graph()
-    G2 = nx.empty_graph()
+def _relabel_graph(G, f=lambda x: -1 * x):
+    """
+    relabels graph nodes by a given relabeling function on integers.
+    By default it flips the sign of the labels (Note this is a self-inverse).
+    Returns the relabeled graph, with the original left intact.
+    """
+    return nx.relabel_nodes(G, {u: f(u) for u in G.nodes()})
 
-    h1 = nx.weisfeiler_lehman_graph_hash(G1)
-    h2 = nx.weisfeiler_lehman_graph_hash(G2)
 
-    assert h1 == h2
+def _relabel_dict(dic, f=lambda x: -1 * x):
+    """
+    relabels dictionary keys by a given relabeling function on integers.
+    By default it flips the sign of the labels (Note this is a self-inverse).
+    Returns the relabeled dictionary, with the original left intact.
+    """
+    return {f(k): v for k, v in dic.items()}
 
 
-def test_relabel():
-    G1 = nx.Graph()
-    G1.add_edges_from(
-        [
-            (1, 2, {"label": "A"}),
-            (2, 3, {"label": "A"}),
-            (3, 1, {"label": "A"}),
-            (1, 4, {"label": "B"}),
-        ]
-    )
-    h_before = nx.weisfeiler_lehman_graph_hash(G1, edge_attr="label")
+class TestFullGraphHash:
+    """Unit tests for the :func:`~networkx.weisfeiler_lehman_graph_hash` function"""
```

Comment:
This class object is never used as a class. Turn all these methods into functions and remove the class test structure.  The functions then have no arguments: e.g.   `def test_empty_graph_hash():`

Reply:
thanks for this pointer, I had assumed classes in pytest were for grouping tests and used it as such, will change